projects/manager/src/app/datasets/dataset-editor/dataset-editor.component.ts

```diff
@@ -38,13 +52,14 @@ export class DatasetEditorComponent implements OnChanges {
     vectorSymbology?: VectorSymbology = undefined;
 
     constructor(
-        private datasetsService: DatasetsService,
-        private workflowsService: WorkflowsService,
+        private readonly datasetsService: DatasetsService,
+        private readonly workflowsService: WorkflowsService,
+        private readonly snackBar: MatSnackBar,
     ) {}
 
     ngOnChanges(changes: SimpleChanges): void {
-        if (changes.datasetName) {
-            this.datasetsService.getDataset(this.datasetName).then((dataset) => {
+        if (changes.datasetListing) {
+            this.datasetsService.getDataset(this.datasetListing.name).then((dataset) => {
```

Comment:
since you use then inside then here:
IMO you can make ngOnChanges async

Reply:
887d120199e497deacab06f0240c8bcb081799f7